bin/reth/src/commands/debug_cmd/build_block.rs

```diff
@@ -198,13 +218,22 @@ impl<C: ChainSpecParser<ChainSpec = ChainSpec>> Command<C> {
                 .await?;
         }
 
+        let is_shanghai_active = provider_factory.chain_spec().is_shanghai_active_at_timestamp(self.timestamp);
+        let withdrawals = if is_shanghai_active {
+            Some(self.parse_withdrawals()?.unwrap_or_default())
+        } else {
+            if self.withdrawals.is_some() {
+                warn!(target: "reth::cli", "Withdrawals provided but Shanghai is not active at timestamp {}", self.timestamp);
+            }
+            None
+        };
+
         let payload_attrs = PayloadAttributes {
             parent_beacon_block_root: self.parent_beacon_block_root,
             prev_randao: self.prev_randao,
             timestamp: self.timestamp,
             suggested_fee_recipient: self.suggested_fee_recipient,
-            // TODO: add support for withdrawals
-            withdrawals: None,
```

Comment:
we def need to set this but perhaps an additional argument for this is overkill and we can just set an empty Some([]) if shanghai is active?